hal/ia32/cpu.c

```diff
@@ -510,10 +540,11 @@ char *hal_cpuFeatures(char *features, unsigned int len)
 	nx &= 0x7fffffff;
 
 	for (p = cpufeatures; p->name != NULL; ++p) {
-		if (p->eax < 0 ? p->eax < -nx : p->eax > nb)
+		if ((p->eax < 0) ? (p->eax < -nx) : (p->eax > nb)) {
```

Comment:
`p->eax` is type of `u32` and is never `< 0`.
Look PR #233.

Reply:
@kemonats thank you for noticing this.

@astalke I'd appreciate if you added a `FIXME` comment indicating an invalid check (lines 543 and 547). We'll remove the `FIXME` note after merging @kemonats pull-request.